src/routes/blog/_query.ts

```diff
@@ -103,6 +113,14 @@ export async function fetchNewsletters(): Promise<Newsletter[]> {
     body: JSON.stringify({ query: newslettersQuery }),
   });
 
-  const newsletters = formatNewsletters(await response.json());
-  return newsletters;
+  let newsletters = formatNewsletters(await response.json());
+  const tags = discernTags(newsletters);
+
+  if (options && options.labels.length > 0) {
+    newsletters = newsletters.filter((post) =>
+      post.labels.some((item) => options.labels.includes(item))
+    );
+  }
+
+  return { tags: tags, posts: newsletters };
```

Comment:
To be fancy, let's use this object construction shorthand syntax:
```suggestion
  return { tags, posts };
```